### PR TITLE
index consumed gas (master)

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
index consumed gas and max gas limit in a block

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/ElrondNetwork/elastic-indexer-go
 go 1.15
 
 require (
-	github.com/ElrondNetwork/elrond-go-core v1.0.1-0.20210721164025-65cf7f169349
+	github.com/ElrondNetwork/elrond-go-core v1.1.1-0.20210915082749-d01a5b3cd3eb
```
Comment (iulianpascalau) [inline go.mod:6]: proper release

File: mock/elasticProcessorStub.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 
 // ElasticProcessorStub -
 type ElasticProcessorStub struct {
-	SaveHeaderCalled                 func(header coreData.HeaderHandler, signersIndexes []uint64, body *block.Body, notarizedHeadersHashes []string, txsSize int) error
+	SaveHeaderCalled                 func(header coreData.HeaderHandler, signersIndexes []uint64, body *block.Body, notarizedHeadersHashes []string, gasConsumptionData indexer.HeaderGasConsumption, txsSize int) error
```
Comment (iulianpascalau) [inline mock/elasticProcessorStub.go:12]: you will change the interface. Should change all implementations

**Review by SebastianMarian — APPROVED** [review]

File: mock/elasticProcessorStub.go (hunk 2 above)
Comment (bogdan-rosianu) [reply mock/elasticProcessorStub.go:12]: what do you mean? this interface is elastic-indexer-go specific. I've changed each stub and implementation

File: go.mod [hunk 3]
```diff
@@ -3,7 +3,7 @@ module github.com/ElrondNetwork/elastic-indexer-go
 go 1.15
 
 require (
-	github.com/ElrondNetwork/elrond-go-core v1.1.1-0.20210915082749-d01a5b3cd3eb
+	github.com/ElrondNetwork/elrond-go-core v1.1.1-0.20210917153134-d6e10327a8c0
```
Comment (iulianpascalau) [inline go.mod:6]: proper release

File: dataParser.go [hunk 4]
```diff
@@ -78,6 +78,7 @@ func (dp *dataParser) getSerializedElasticBlockAndHeaderHash(
 		PrevHash:              hex.EncodeToString(header.GetPrevHash()),
 		SearchOrder:           computeBlockSearchOrder(header),
 		GasConsumed:           gasConsumptionData.GasConsumed,
+		GasRefunded:           gasConsumptionData.GasRefunded,
```
Comment (iulianpascalau) [inline dataParser.go:81]: do we need to have these changes also on the new indexer branch?

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: go.mod (hunk 3 above)
Comment (raduchis) [reply go.mod:6]: :+1:

File: dataParser.go (hunk 4 above)
Comment (bogdan-rosianu) [reply dataParser.go:81]: I have already integrated it in PR #58 - might add the gas refunded there as well

**Review by raduchis — APPROVED** [review]

File: mock/elasticProcessorStub.go (hunk 2 above)
Comment (iulianpascalau) [reply mock/elasticProcessorStub.go:12]: right, the ElasticProcessor is just a subcomponent used in the indexer implementation and in one integration test in elrond-go

**Review by iulianpascalau — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Pull request merged**